src/main/java/games/strategy/engine/lobby/server/userDB/DBUserController.java

```diff
@@ -1,210 +1,112 @@
+ */
+/*
+ * TODO: datasource migration note, from derby (java DB) to MySQL
+ *  This class is currently set up to write and read from multiple data sources.
+ *  This is a migration tool, to go from one data source to another. When we switch over completely
+ *  we can simplify and write/read to the primary datasource directly.
+ */
+public class DBUserController implements UserDao {
+  private static final Logger logger = Logger.getLogger(DBUserController.class.getName());
 
-public class DBUserController {
-  private static final Logger s_logger = Logger.getLogger(DBUserController.class.getName());
+  private static final Collection<UserDaoPrimarySecondary> implementations = Arrays.asList(
+      new DerbyUserController(),
+      new SqlUserController()
+  );
 
   /**
-   * @return if this user is valid.
+   * Stores a convenience reference to the authoritative DAO implementation.
+   * We use this datasource for reading, we always write to it, any errors during read/write are considered
+   * critical (on the other hand secondary DAO read/write errors can be logged and ignored)
    */
-  public String validate(final String userName, final String email, final String hashedPassword) {
-    if (email == null || !Util.isMailValid(email)) {
-      return "Invalid email address";
-    }
-    if (hashedPassword == null || hashedPassword.length() < 3 || !hashedPassword.startsWith(MD5Crypt.MAGIC)) {
-      return "Invalid password";
-    }
-    return validateUserName(userName);
-  }
+  private static final UserDao primaryDao;
 
-  public static String validateUserName(final String userName) {
-    // is this a valid user?
-    if (userName == null || !userName.matches("[0-9a-zA-Z_-]+") || userName.length() <= 2) {
-      return "Usernames must be at least 3 characters long and can only contain alpha numeric characters, -, and _";
-    }
-    if (userName.contains(InGameLobbyWatcher.LOBBY_WATCHER_NAME)) {
-      return InGameLobbyWatcher.LOBBY_WATCHER_NAME + " cannot be part of a name";
-    }
-    if (userName.toLowerCase().contains("admin")) {
-      return "Username can't contain the word admin";
-    }
-    return null;
+
+  static {
+    primaryDao = determinePrimaryDataSource();
   }
 
-  public static void main(final String[] args) throws SQLException {
-    Database.getConnection().close();
+
+  private static UserDao determinePrimaryDataSource() {
+    List<UserDao> primaryDataSourceCandidates = implementations.stream()
+        .filter(UserDaoPrimarySecondary::isPrimary)
+        .collect(Collectors.toList());
+
+    Preconditions.checkState(
+        primaryDataSourceCandidates.size() == 1,
+        "Startup Assumption - exactly one UserDao implementation is 'primary'");
+
+    return primaryDataSourceCandidates.get(0);
   }
 
+
   /**
    * @return null if the user does not exist.
    */
+  @Override
   public String getPassword(final String userName) {
-    final String sql = "select password from ta_users where username = ?";
```

Comment:
The sql code for the most part was all moved verbatim to `DerbyUserController`. By doing this we're allowing DBUserController to be a facade that does multiple writes/reads. This choice does involve a good chunk of code moving around, but we will keep three things in isolation:
- legacy DB access code
- new DB access code
- handling of legacy vs primary DB access code